lighthouse-core/lib/url-shim.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * URL shim so we keep our code DRY
+ */
+
+'use strict';
+
+module.exports = self.URL || require('url').URL || require('whatwg-url').URL;
```

Comment:
since this is in its own file now, you can do something more verbose like `module.exports = (self && self.URL) || require('url').URL || require('whatwg-url').URL;` and eliminate the need for the `global.self = {};` lines in tests and the global `self` in the .eslintrc (should just need a local `/* global self */` here)